google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -700,6 +867,36 @@ def _get_initial_request(
         # Return the initial request.
         return request
 
+    def _send_lease_modacks(self, ack_ids: Sequence[str], ack_deadline: int):
+        exactly_once_enabled = False
+        with self._exactly_once_enabled_lock:
+            exactly_once_enabled = self._exactly_once_enabled
+        if exactly_once_enabled:
+            items = []
+            for ack_id in ack_ids:
+                future = futures.Future()
+                request = requests.ModAckRequest(ack_id, ack_deadline, future)
+                items.append(request)
+
+            assert self._dispatcher is not None
+            self._dispatcher.modify_ack_deadline(items)
+
+            for req in items:
+                try:
+                    req.future.result()
```

Comment:
This will block right? Any concerns around performance?

Reply:
Ack and modack RPCs calls have always been synchronous in this code path so this doesn't change that fact. So we're fine.